packages/typescript/test/test.js

```diff
@@ -359,6 +359,29 @@ test('complies code that uses browser functions', async (t) => {
   t.true(code.includes('navigator.clipboard.readText()'), code);
 });
 
+test('reports compiler options errors', async (t) => {
+  process.chdir('fixtures/invalid-compiler-options');
+  const warnings = [];
+  const err = await t.throwsAsync(() =>
+    rollup({
+      input: 'main.ts',
+      plugins: [
+        typescript({
+          tsconfig: 'tsconfig.json'
+        })
+      ],
+      onwarn({ toString, ...warning }) {
+        warnings.push(warning);
+      }
+    })
+  );
+
+  t.true(
+    err.message.includes('Composite projects may not disable declaration emit'),
```

Comment:
Should we override this flag too?

Reply:
I think we should add `composite` to our list of overridden compiler options if it causes an error.

Reply:
I agree - I think this is best done in a separate PR however. It seems ideal to output compilerOptions errors, regardless of the overridden compiler options - Future typescript versions may introduce new compilerOption errors. 

(And thinking about it, when I do this in a separate PR, it'll break this test)